### PR TITLE
Add LoadProjectAsync to the banned symbol list

### PR DESCRIPTION
**Pull request by davkean**

### CODE REVIEW

File: src/BannedSymbols.txt [hunk 1]
```diff
@@ -6,6 +6,8 @@ T:Microsoft.VisualStudio.ProjectSystem.VS.IProjectGuidService2; Using ISafeProje
 P:Microsoft.VisualStudio.ProjectSystem.VS.IProjectAsyncLoadDashboard.ProjectLoadedInHost; Using IUnconfiguredProjectTasksService.ProjectLoadedInHost prevents waiting indefinitely when the project is closed before it has finished loading
 
 M:Microsoft.VisualStudio.ProjectSystem.CommonProjectSystemTools.LoadedProject(Microsoft.VisualStudio.ProjectSystem.IProjectAsynchronousTasksService); Using IUnconfiguredProjectTasksService.LoadedProjectAsync is unit testable
+M:Microsoft.VisualStudio.ProjectSystem.CommonProjectSystemTools.LoadedProjectAsync(Microsoft.VisualStudio.ProjectSystem.IProjectAsynchronousTasksService,System.Func{System.Threading.Tasks.Task},System.Boolean); Using IUnconfiguredProjectTasksService.LoadedProjectAsync is unit testable
+M:Microsoft.VisualStudio.ProjectSystem.CommonProjectSystemTools.LoadedProjectAsync`1(Microsoft.VisualStudio.ProjectSystem.IProjectAsynchronousTasksService,System.Func{System.Threading.Tasks.Task{``0}},System.Boolean); Using IUnconfiguredProjectTasksService.LoadedProjectAsync is unit testable
```
Comment (drewnoakes) [inline src/BannedSymbols.txt]: Is it worth mentioning `UnconfiguredProjectTasksService` as a replacement in the messages? Also on line 8.

File: src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/Subscriptions/DependencyRulesSubscriber.cs [hunk 2]
```diff
@@ -186,11 +186,6 @@ private IReadOnlyCollection<string> GetWatchedRules(RuleHandlerType handlerType)
 
             await _tasksService.LoadedProjectAsync(async () =>
             {
-                if (_tasksService.UnloadCancellationToken.IsCancellationRequested)
-                {
-                    return;
```
Comment (drewnoakes) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/Subscriptions/DependencyRulesSubscriber.cs]: Was this doing anything useful? Doesn't `LoadedProjectAsync` prevent unloading during the callback anyway?

File: src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/Subscriptions/DependencySharedProjectsSubscriber.cs [hunk 3]
```diff
@@ -108,8 +108,6 @@ private async Task OnProjectChangedAsync(Tuple<IProjectSubscriptionUpdate, IProj
 
             await _tasksService.LoadedProjectAsync(() =>
             {
-                _tasksService.UnloadCancellationToken.ThrowIfCancellationRequested();
```
Comment (drewnoakes) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/Subscriptions/DependencySharedProjectsSubscriber.cs]: Ditto.

File: src/BannedSymbols.txt (hunk 1 above)
Comment (davkean) [reply src/BannedSymbols.txt]: UnconfiguredProjectTasksService is the implementation, but you import IUnconfiguredProjectTasksService.

File: src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/Subscriptions/DependencyRulesSubscriber.cs (hunk 2 above)
Comment (davkean) [reply src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/Subscriptions/DependencyRulesSubscriber.cs]: This line is effectively what LoadProjectAsync does before calling this callback, it won't start the callback if this is true.

**Pull request merged**